server/asset/eth/eth.go

```diff
@@ -106,13 +120,15 @@ var _ asset.Backend = (*Backend)(nil)
 // before use.
 func unconnectedETH(logger dex.Logger, cfg *config) *Backend {
 	ctx, cancel := context.WithCancel(context.Background())
+	mainnetContractAddr := common.HexToAddress(mainnetContractAddr)
```

Comment:
For a first go, we'd at least want this switching the contract address based on the network, no?